jsonpointer.py

```diff
@@ -153,28 +153,53 @@ def to_last(self, doc):
         if not self.parts:
             return doc, None
 
-        for part in self.parts[:-1]:
-            doc = self.walk(doc, part)
-
-        return doc, self.get_part(doc, self.parts[-1])
-
-
-    def resolve(self, doc, default=_nothing):
+        doc = self.resolve(doc, parts=self.parts[:-1])
+        last = self.parts[-1]
+        ptype = type(doc)
+        if ptype == dict:
+            pass
+        elif ptype == list or isinstance(doc, Sequence):
```

Comment:
OK, so in that case the ``is`` operator should be used, and it should have some comment explaining that this is a performance optimization.